independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/decorators/validation/NonDependentScopeTest.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.arc.test.decorators.validation;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.quarkus.arc.test.ArcTestContainer;
+import javax.annotation.Priority;
+import javax.decorator.Decorator;
+import javax.decorator.Delegate;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class NonDependentScopeTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(Converter.class, DecoratorWithWrongScope.class).shouldFail().build();
+
+    @Test
+    public void testFailure() {
+        assertNotNull(container.getFailure());
+        assertTrue(
+                container.getFailure().getMessage()
+                        .contains("DecoratorWithWrongScope must be @Dependent: javax.enterprise.context.ApplicationScoped"),
+                container.getFailure().getMessage());
+    }
+
+    interface Converter<T, U> {
+
+        T convert(T value);
```

Comment:
One of the `T`s here should probably be `U` :rofl:

Reply:
Yeah, but that's not the point of this test, right? :D

Reply:
Sure, I don't really understand decorators, so I have to look for things like this :laughing: